src/main/webapp/resources/js/pages/analysis/components/AnalysisExcelPreview.jsx

```diff
@@ -20,16 +21,27 @@ const ExcelOutputWrapper = styled.div`
 `;
 
 export default function AnalysisExcelPreview({ output }) {
-  const [excelHeaders, setExcelHeaders] = useState([]);
-  const [excelRows, setExcelRows] = useState([]);
-  const [excelSheetNames, setExcelSheetNames] = useState([]);
+  const [excelHeaders, setExcelHeaders] = React.useState([]);
+  const [excelRows, setExcelRows] = React.useState([]);
+  const [excelSheetNames, setExcelSheetNames] = React.useState([]);
   const MAX_TABLE_ROWS_PER_PAGE = 5;
-  const [currSheetIndex, setCurrSheetIndex] = useState("0");
-  const [parseError, setParseError] = useState(null);
+  const [currSheetIndex, setCurrSheetIndex] = React.useState("0");
+  const [parseError, setParseError] = React.useState(null);
 
-  useEffect(() => {
-    parseExcel(output.analysisSubmissionId, output.filename, currSheetIndex).then(({data}) => {
-      if(!data.parseError) {
+  const [total, setTotal] = React.useState(0);
+  const [paginationOptions, setPaginationOptions] = React.useState(null);
+
+  React.useMemo(() => {
```

Comment:
This should be:

```jsx
const paginationOptions = React.useMemo(() => {
  getPaginationOptions(total);
}, [total]);
```

Reply:
Updated in c96bd77d27718c9064e6535c7aac61dd11b3f80b